pkg/providers/v1/tags.go

```diff
@@ -154,6 +154,31 @@ func (t *awsTagging) hasClusterTag(tags []ec2types.Tag) bool {
 	return false
 }
 
+// hasClusterTagOwned checks if the resource has the cluster tag with the value "owned"
+// This is used to determine if the resource is owned by the controller.
+// It checks both legacy and new tags.
+func (t *awsTagging) hasClusterTagOwned(tags []ec2types.Tag) bool {
+	if len(t.ClusterID) == 0 {
+		return false
```

Comment:
is there a possibility that this could hide a false negative? or should this be an error of some sort?

Reply:
good point! It could return false negative if ClusterID could assume empty values. Looks like the controller will not initialize if ClusterID is empty:
https://github.com/kubernetes/cloud-provider-aws/blob/5f62c03b684933f64705e0223f4905e8090cbc6e/pkg/providers/v1/tags.go#L65-L83

Anyway, it would be a good idea returning an error making this function more consistent. I will update in the next changes. Thanks!

Reply:
Hey @kmala do you have any use case in mind that expects empty clusterID? I am leaning towards raising an error when empty here, making it consistent with tagging validations.

Reply:
thanks @mtulio , i'm not as familiar with the underlying code so i just wanted to make sure there wasn't a possible error here.